src/Treemap.js

```diff
@@ -37,13 +38,20 @@ export default class Treemap extends Viz {
   }
 
   /**
-      Extends the draw behavior of the abstract Viz class.
+      @memberof Treemap
+      @desc Extends the draw behavior of the abstract Viz class.
       @private
   */
   _draw(callback) {
 
     super._draw(callback);
 
+    // Sort input data and assign index property to them.
+    const sortedFilteredData = this._filteredData.sort(this._sort);
+    for (let i = 0; i < sortedFilteredData.length; ++i) {
+      sortedFilteredData[i].index = i;
+    }
+    
```

Comment:
This logic is actually incorrect, but only seems to work because you are testing data where the `sum` accessor is also set to use `"value"`. There is a distinct difference between the number that the user defines for sizing, and the internal "value" property that the treemap layout calculates. The `sort` function actually refers to the internal treemap layout's value.

Take a look at the work I do below with `shapeData`, which flattens the treemap layout into an array of nodes. That array can be sorted using the `this._sort` function, and then the index of node in that array can be used for ranking. For example:
```js
this._rankData = shapeData.sort(this._sort).map(d => d.data);
```

Reply:
please remove these lines of commented code